spec/api-desktop-capturer-spec.js

```diff
@@ -91,12 +91,145 @@ describe('desktopCapturer', () => {
     w.show()
     await wShown
 
-    const sources = await desktopCapturer.getSources({ types: ['window', 'screen'], thumbnailSize: { width: 0, height: 0 } })
+    const sources = await desktopCapturer.getSources({
+      types: ['window', 'screen'],
+      thumbnailSize: { width: 0, height: 0 }
+    })
     w.destroy()
     expect(sources).to.be.an('array').that.is.not.empty()
     for (const { thumbnail: thumbnailImage } of sources) {
       expect(thumbnailImage).to.be.a('NativeImage')
       expect(thumbnailImage.isEmpty()).to.be.true()
     }
   })
+
+  it('getMediaSourceId should match DesktopCapturerSource.id', async () => {
+    const { BrowserWindow } = remote
+    const w = new BrowserWindow({ show: false, width: 100, height: 100 })
+    const wShown = emittedOnce(w, 'show')
+    const wFocused = emittedOnce(w, 'focus')
+    w.show()
+    w.focus()
+    await wShown
+    await wFocused
+
+    const mediaSourceId = w.getMediaSourceId()
+    const sources = await desktopCapturer.getSources({
+      types: ['window'],
+      thumbnailSize: { width: 0, height: 0 }
+    })
+    w.destroy()
+
+    // TODO(julien.isorce): investigate why |sources| is empty on the linux
+    // bots while it is not on my workstation, as expected, with and without
+    // the --ci parameter.
+    if (process.platform === 'linux' && sources.length === 0) { return }
+
+    expect(sources).to.be.an('array').that.is.not.empty()
+    const foundSource = sources.find((source) => {
+      return source.id === mediaSourceId
+    })
+    expect(mediaSourceId).to.equal(foundSource.id)
+  })
+
+  it('moveAbove should move the window at the requested place', async () => {
+    // DesktopCapturer.getSources() is guaranteed to return in the correct
+    // z-order from foreground to background.
+    const MAX_WIN = 4
+    const { BrowserWindow } = remote
+    const mainWindow = remote.getCurrentWindow()
+    const wList = [mainWindow]
+    // Add MAX_WIN-1 more window so we have MAX_WIN in total.
+    for (let i = 0; i < MAX_WIN - 1; i++) {
+      const w = new BrowserWindow({ show: true, width: 100, height: 100 })
+      wList.push(w)
+      w.loadURL('about:blank')
+      w.show()
```

Comment:
is the `show()` necessary since the window is constructed with `show: true`?

(also `loadURL('about:blank')` seems unnecessary)

Reply:
Done, removed